### DISCUSSION TITLE
Waiting for exchange. Make sure you're using the Urql Devtools exchange! ERR_UNKNOWN_URL_SCHEME

### DISCUSSION THREAD

**Discussion by karthickthankyou** [urql-graphql/urql#2132] [Q&A] [answered]
The devtools says "Waiting for exchange. Make sure you're using the Urql Devtools exchange!" With status code of 404 net::ERR_UNKNOWN_URL_SCHEME.
<img width="588" alt="Screenshot 2021-12-07 at 8 04 44 PM" src="https://user-images.githubusercontent.com/13064502/145048443-b1a8f0c1-a8bc-4770-ab92-afb050cc4c74.png">

urql version & exchanges:

**Steps to reproduce**

1. Install "@urql/devtools"
2. Initialize urql client with the devtools exchange.
```
import { ssrExchange } from 'urql'
import { devtoolsExchange } from '@urql/devtools'
import { initUrqlClient } from 'next-urql'
import { defaultExchanges } from 'urql'

export const ssrCache = ssrExchange({ isClient: false })
export const client = initUrqlClient(
  {
    url: 'https://rickandmortyapi.com/graphql',
    exchanges: [devtoolsExchange, ...defaultExchanges],
  },
  true,
)

export default client
```

Versions:
```
"dependencies": {
    "@apollo/client": "^3.5.5",
    "graphql": "^16.0.1",
    "graphql-codegen-apollo-next-ssr": "^1.7.1",
    "next": "12.0.7",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  },
  "devDependencies": {
    "@graphql-codegen/cli": "^2.3.0",
    "@graphql-codegen/typescript": "^2.4.1",
    "@graphql-codegen/typescript-operations": "^2.2.1",
    "@graphql-codegen/typescript-react-apollo": "^3.2.2",
    "@types/graphql": "^14.5.0",
    "@types/node": "16.11.11",
    "@types/react": "17.0.37",
    "apollo": "^2.33.9",
    "eslint": "8.4.0",
    "eslint-config-next": "12.0.7",
    "typescript": "4.5.2"
  }
```

The example codesandbox project https://codesandbox.io/s/urql-issue-template-client-iui0o is also showing the same error. 

I've tried the below.

<img width="759" alt="Screenshot 2021-12-07 at 9 07 25 PM" src="https://user-images.githubusercontent.com/13064502/145059444-e15c5b0c-a9a5-41a9-976e-ee8b35c94b48.png">

Thank you.

**Comment by JoviDeCroock** [comment, marked as answer]
The reproduction doesn't use the devtoolsExchange, that being said, if you're using next you might want to add it in `withUrqlClient` rather than `initUrqlClient`

After altering your sandbox to https://codesandbox.io/s/async-hooks-630mr?file=/src/index.js it works correctly when surfing to its url https://630mr.csb.app/ and opening the devtools
